executor/executor_simple.go

```diff
@@ -343,8 +345,39 @@ func (e *SimpleExec) executeCreateUser(s *ast.CreateUserStmt) error {
 	return nil
 }
 
+func (e *SimpleExec) executeDropUser(s *ast.DropUserStmt) error {
+	failedUsers := make([]string, 0, len(s.UserList))
+	for _, user := range s.UserList {
+		userName, host := parseUser(user)
+		exists, err := userExists(e.ctx, userName, host)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if !exists {
+			if !s.IfExists {
+				failedUsers = append(failedUsers, user)
+			}
+			continue
+		}
+		sql := fmt.Sprintf(`DELETE FROM %s.%s WHERE Host = "%s" and User = "%s";`, mysql.SystemDB, mysql.UserTable, host, userName)
+		_, err = e.ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(e.ctx, sql)
+		if err != nil {
+			failedUsers = append(failedUsers, user)
+		}
+	}
+	err := e.ctx.CommitTxn()
```

Comment:
If we drop user1 , user3 successful but fail to drop user2, the txn should be committed and return an error at the same time. If we don't commit it manually, it will rollback instead.